worker/upload.go

```diff
@@ -918,6 +936,14 @@ func (u *uploader) execute(req *sectorUploadReq, rev types.FileContractRevision)
 	span.SetAttributes(attribute.Int64("duration", elapsed.Milliseconds()))
 	span.RecordError(err)
 	span.End()
+
+	// update bus
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	if err := u.mgr.b.AddUploadedSector(ctx, req.upload.id, fcid, root); err != nil {
```

Comment:
This needs to happen before uploading the sector. Otherwise we leave a small window where the sector was uploaded but not added to the bus in which another worker might prune the sector from the contract.

Reply:
Yeah I realised the small window was there but didn't want to optimistically persist sectors without removing them on failure. Did it anyway though, can't have the small window at all idd